google/resumable_media/requests/download.py

```diff
@@ -58,12 +62,43 @@ def _write_to_stream(self, response):
 
         Args:
             response (~requests.Response): The HTTP response object.
+
+        Raises:
+            DataCorruption: If the download's checksum doesn't agree with
+                server-computed checksum.
         """
+        md5_hash = hashlib.md5()
+        # TODO: Add support for validating CRC32C for composite objects. Note
+        # that this is somewhat painful in Python 2 because most OS distros of
+        # Python 2 don't include a compiled crcmod (and executing without a
+        # compiled crcmod is very slow; and installing a compiled crcmod will
+        # make getting google-cloud-python working well more difficult).
+        expected_md5_hash = None
+        if ('X-Goog-Hash' in response.headers
+            and response.headers['X-Goog-Hash']):
+            for checksum in response.headers['X-Goog-Hash'].split(','):
+                name, value = checksum.split('=', 1)
+                if name == 'md5':
+                    expected_md5_hash = value
+        if not expected_md5_hash:
+            logging.getLogger().info(
```

Comment:
Instead we should make this logger a non-public attribute of the module.

Reply:
Done